vllm/model_executor/models/mlp_speculator.py

```diff
@@ -168,6 +169,11 @@ def generate_proposals(
 
         for head_index in range(num_predict_tokens):
 
+            if current_platform.is_hpu():
+                last_tokens_batches = last_tokens.shape[0]
+                previous_hidden_states_batches = previous_hidden_states.shape[0]
+                if (last_tokens_batches > previous_hidden_states_batches):
+                    last_tokens = last_tokens[:previous_hidden_states_batches]
```

Comment:
Based on previous log:
last_tokens_batches is num_tokens of this batch
previous_hidden_states_batches is indexed https://github.com/HabanaAI/vllm-fork/blame/habana_main/vllm/sequence.py#L1341, which leads the shape might be smaller than last_tokens_batches

But I don't think we have prove point saying cut tail is correct since we don't know if in sequence.py, it will select from middle.

I still just you should fix the shape in https://github.com/HabanaAI/vllm-fork/blame/habana_main/vllm/sequence.py